### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by mhsmith**
Using the same approach as https://github.com/beeware/briefcase/pull/915.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by mhsmith** [comment]
@freakboy3742: Looks like this repository isn't set up with `secrets.TEST_PYPI_PASSWORD` and `secrets.PYPI_PASSWORD`. Let me know once you've done that, and I'll complete the testing.

**Comment by freakboy3742** [comment]
@mhsmith So - there could be a complication here. PyPI tokens are either "whole of repository", or "single project". If I create a "Toga"-specific token, we won't be able to use it to upload toga-core, toga-android, etc. The main reason to *not* use a "project wide" token is to limit the exposure of damage if a token is stolen; but without it, we'll need to do every wheel upload with a different token. Thoughts?

**Comment by mhsmith** [comment]
I see you have a lot of things in your [PyPI account](https://pypi.org/user/freakboy3742/), so it's probably best to keep them separate. If you use the following syntax, I think I can rewrite the workflow to use it:
* PYPI_PASSWORD_toga
* PYPI_PASSWORD_toga_android
* etc.

And similarly for TEST_PYPI_PASSWORD.

**Comment by freakboy3742** [comment]
I've set up tokens for each individual project, following the `PYPI_PASSWORD_TOGA_FOO` pattern (the capitalisation is enforced by Github).

**Review by mhsmith — COMMENTED** [review]
Aside from updating the workflows, this PR also adds a "src" directory within all the packages under test. As discussed [here](https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html), this allows the tests to be run against an installed package without interference from the original source code directory being on `sys.path`.

As part of the testing, I created 0.0.1rc5 and 0.0.1rc6 releases on GitHub and PyPI.

**Review by freakboy3742 — APPROVED** [review]
There's some minor things that we could tweak/improve, but given that (a) we're about to make a bunch more changes to the high level project structure, and (b) we've got the minor deadline of DjangoCon sprints, I'm happy with this as is.

File: tox.ini [hunk 1]
```diff
@@ -59,7 +56,7 @@ deps =
     wheel
     twine
 whitelist_externals =
-    release.sh
-changedir = {toxinidir}
+    bash
 commands =
-    ./release.sh package
+    # Bash scripts can't be executed directly on Windows.
```
[the remark this answers is not in the document]
Comment (mhsmith) [reply tox.ini:61]: You need to provide your own copy of Bash, of course, but aside from that it all seems fine.

**Pull request merged**